easybuild/easyconfigs/p/PyTorch/PyTorch-1.12.0-foss-2022a-CUDA-11.7.0.eb

```diff
@@ -0,0 +1,127 @@
+name = 'PyTorch'
+version = '1.12.0'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://pytorch.org/'
+description = """Tensors and Dynamic neural networks in Python with strong GPU acceleration.
+PyTorch is a deep learning framework that puts Python first."""
+
+toolchain = {'name': 'foss', 'version': '2022a'}
+
+sources = [{
+    'filename': '%(name)s-%(version)s.tar.gz',
+    'git_config': {
+        'url': 'https://github.com/pytorch',
+        'repo_name': 'pytorch',
+        'tag': 'v%(version)s',
+        'recursive': True,
+    },
+}]
+patches = [
+    '%(name)s-1.7.0_avoid-nan-in-test-torch.patch',
+    '%(name)s-1.7.0_disable-dev-shm-test.patch',
+    '%(name)s-1.8.1_dont-use-gpu-ccc-in-test.patch',
+    '%(name)s-1.9.0_limit-world-size-for-zero-redundancy-opt-test.patch',
+    '%(name)s-1.10.0_fix-test-dataloader-fixed-affinity.patch',
+    '%(name)s-1.10.0_skip_cmake_rpath.patch',
+    '%(name)s-1.11.0_increase-distributed-test-timeout.patch',
+    '%(name)s-1.11.0_increase_c10d_gloo_timeout.patch',
+    '%(name)s-1.11.0_disable_failing_jit_cuda_fuser_tests.patch',
+]
+checksums = [
+    None,  # PyTorch-1.12.0.tar.gz
+    'b899aa94d9e60f11ee75a706563312ccefa9cf432756c470caa8e623991c8f18',  # PyTorch-1.7.0_avoid-nan-in-test-torch.patch
+    '622cb1eaeadc06e13128a862d9946bcc1f1edd3d02b259c56a9aecc4d5406b8a',  # PyTorch-1.7.0_disable-dev-shm-test.patch
+    '89ac7a8e9e7df2e64cf8404fe3a279f5e9b759fee41c9de3aaff9c22f385c2c6',  # PyTorch-1.8.1_dont-use-gpu-ccc-in-test.patch
+    # PyTorch-1.9.0_limit-world-size-for-zero-redundancy-opt-test.patch
+    'ff573660913ce055e24cfd194ce747ba5685091c631cfd443eae2a99d56b57ea',
+    # PyTorch-1.10.0_fix-test-dataloader-fixed-affinity.patch
+    '313dca681f45ce3bc7c4557fdcdcbe0b77216d2c708fa30a2ec0e22c44876707',
+    'ac05943bb205623f91ef140aa00869efc5fe844184bd666bebf5405808610448',  # PyTorch-1.10.0_skip_cmake_rpath.patch
+    # PyTorch-1.11.0_increase-distributed-test-timeout.patch
+    '087ad20163a1291773ae3457569b80523080eb3731e210946459b2333a919f3f',
+    # PyTorch-1.11.0_increase_c10d_gloo_timeout.patch
+    '20cd4a8663f74ab326fdb032b926bf5c7e94d9750c515ab9050927ba00cf1953',
+    # PyTorch-1.11.0_disable_failing_jit_cuda_fuser_tests.patch
+    'e7bfe120a8b3fe2b40dac6839852a5fbab3cb3429fbe44a0fc3a1800adaaee51',
+]
+
+builddependencies = [
+    ('CMake', '3.23.1'),
+    ('hypothesis', '6.46.7'),
+]
+dependencies = [
+    ('CUDA', '11.7.0', '', True),
+    ('Ninja', '1.10.2'),  # Required for JIT compilation of C++ extensions
+    ('Python', '3.10.4'),
+    ('protobuf', '3.19.4'),
+    ('protobuf-python', '3.19.4'),
+    ('pybind11', '2.9.2'),
+    ('SciPy-bundle', '2022.05'),
+    ('PyYAML', '6.0'),
+    ('MPFR', '4.1.0'),
+    ('GMP', '6.2.1'),
+    ('numactl', '2.0.14'),
+    ('FFmpeg', '5.0.1'),
```

Comment:
@casparvl Does PyTorch require `FFmpeg` 5.x? If not, why not stick to FFmpeg 4.x for now, since that's likely going to be the version that gets the upper hand for a while still?

Reply:
Well, we have several PRs already based on 5.X: 

- https://github.com/easybuilders/easybuild-easyconfigs/pull/16162
- https://github.com/easybuilders/easybuild-easyconfigs/pull/16276
- This one

Some of these PRs have been around for a while, and by now, we at least rolled out this PR and the one for LAMMPS on our own systems, so at least for us switching back to 4.X would be a bit annoying (we prefer not to redeploy stuff unless we have to for reproducibility).

Apart from `ParaView` (https://github.com/easybuilders/easybuild-easyconfigs/pull/15938#issuecomment-1200509240) we haven't seen issues so far, at least in building the software. For runtime issues: the newer modules probably haven't been that widely used yet, so it's harder to say.

I guess the main question is: how important do we find compatibility between `ParaView` and other modules? I did a quick look, in 2021a, it seems `FFmpeg` was a dependency for `PyTorch`, `ParaView`, `OpenCV` and `LAMMPS`. Thus, it seems that with the current PRs, we've basically already covered all usage of `FFmpeg` as a dependency.

Reply:
in the last EB conf call, it was agreed to stick to 4.4.2 for 2022a

Reply:
Done